lib/features/mailbox/presentation/mailbox_controller.dart

```diff
@@ -100,22 +93,12 @@ class MailboxController extends BaseController {
       BuildContext context,
       PresentationMailbox presentationMailboxSelected
   ) {
-    selectedMailbox.value = presentationMailboxSelected;
-
-    _updateMailboxCurrentInMailboxDashBoard(presentationMailboxSelected);
+    Get.find<MailboxDashBoardController>().setSelectedMailbox(presentationMailboxSelected);
+    Get.find<MailboxDashBoardController>().setSelectedEmail(PresentationEmail.presentationEmailEmpty);
 
+    // Close MenuLeft in device mobile
     if (responsiveUtils.isMobile(context)) {
-      final mailboxDashBoardController = Get.find<MailboxDashBoardController>();
-      mailboxDashBoardController.scaffoldKey.currentState?.openEndDrawer();
-    }
-
-    goToThreadView();
-  }
-
-  void goToThreadView() {
-    if (Get.isRegistered<ThreadController>()) {
-      final threadController = Get.find<ThreadController>();
-      threadController.updateMailbox(presentationMailbox: selectedMailbox.value, accountId: accountId);
+      Get.find<MailboxDashBoardController>().scaffoldKey.currentState?.openEndDrawer();
```

Comment:
please create a method inside `MailboxDashBoardController` `openEndDrawer()`. Should not public `scaffoldKey`